v2/middlewares/measurement.go

```diff
@@ -0,0 +1,69 @@
+package middlewares
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+
+	"github.com/honestbank/kp/v2/internal/middleware"
+)
+
+var operationDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+	Name:    "kp_operation_time_milliseconds",
+	Buckets: []float64{1, 5, 50, 200, 500, 1_000, 2_000, 5_000, 15_000, 45_000},
+}, []string{"result", "error"})
+
+var totalProcessed = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Name: "message_processed_total",
```

Comment:
From the naming of the metric I see most of the metrics named rather as 
```suggestion
		Name: "message_processed",
```
Because prometheus knows this is a counter, it is usually not necessary add the `_total`. But not a big deal if you add it.

Reply:
removed the whole counter according to our discussion.